src/openvic-simulation/pop/Pop.cpp

```diff
@@ -97,27 +192,91 @@ bool PopManager::add_pop_type(
 		return false;
 	}
 	if (sprite <= 0) {
-		Logger::error("Invalid pop type sprite index for ", identifier, ": ", sprite);
+		Logger::error("Invalid pop type sprite index for ", identifier, ": ", sprite, " (must be positive)");
+		return false;
+	}
+	if (max_size <= 0) {
+		Logger::error("Invalid pop type max size for ", identifier, ": ", max_size, " (must be positive)");
+		return false;
+	}
+	if (merge_max_size <= 0) {
+		Logger::error("Invalid pop type merge max size for ", identifier, ": ", merge_max_size, " (must be positive)");
 		return false;
 	}
-	if (max_size < 0) {
-		Logger::error("Invalid pop type max size for ", identifier, ": ", max_size);
+
+	if (research_points < 0) {
+		Logger::error("Invalid pop type research points for ", identifier, ": ", research_points, " (cannot be negative)");
 		return false;
 	}
-	if (merge_max_size < 0) {
-		Logger::error("Invalid pop type merge max size for ", identifier, ": ", merge_max_size);
+	if (leadership_points < 0) {
+		Logger::error("Invalid pop type leadership points for ", identifier, ": ", leadership_points, " (cannot be negative)");
 		return false;
 	}
+	if (research_leadership_optimum < 0) {
+		Logger::error(
+			"Invalid pop type research/leadership optimum for ", identifier, ": ", research_leadership_optimum,
+			" (cannot be negative)"
+		);
+		return false;
+	}
+	if ((research_points > 0 || leadership_points > 0) != (research_leadership_optimum > 0)) {
```

Comment:
```suggestion
	if ((research_points != 0 || leadership_points != 0) != (research_leadership_optimum > 0)) {
```
Only non-zero matters here. Other checks already handle negative values.